src/components/Errors/GlobalErrorsList/DaysFilter/index.tsx

```diff
@@ -0,0 +1,158 @@
+  const [selectedDays, setSelectedDays] = useState<number>();
+  const [currentValue, setCurrentValue] = useState<number>();
+  const previousSelectedDays = usePrevious(selectedDays);
+  const handleSelectionChange = useCallback(
+    (days: number) => {
+      const value = selectedDays === days ? undefined : days;
+      setSelectedDays(value);
+      setCurrentValue(value);
+      setIsDateMenuOpen(false);
+    },
+    [selectedDays]
+  );
+
+  const daysFilterMenuItems = useMemo(
+    () =>
+      Object.values(DEFAULT_LIST_OPTIONS).map((x) => ({
+        id: x.toString(),
+        label: "Last " + getOptionLabel(x),
+        isSelected: selectedDays === x,
+        onClick: () => handleSelectionChange(x)
+      })),
+    [handleSelectionChange, selectedDays]
+  );
+
+  useEffect(() => {
+    if (previousSelectedDays !== selectedDays) {
+      onChanged(selectedDays);
+    }
+  }, [selectedDays, previousSelectedDays, onChanged]);
+
+  const handleSortingMenuButtonClick = () => {
+    sendUserActionTrackingEvent(
+      trackingEvents.GLOBAL_ERRORS_VIEW_DATES_FILTERS_CHANGE
+    );
+    setIsDateMenuOpen(!isDateMenuOpen);
+    setCurrentValue(selectedDays);
+  };
+
+  const handleCounterInputChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const newValue = e.target.value;
+    const intValue = parseInt(newValue);
+    const days =
+      !newValue || isNaN(intValue)
+        ? undefined
+        : intValue > MAX_VALUE
+        ? selectedDays
+        : intValue;
+
+    sendUserActionTrackingEvent(
+      trackingEvents.GLOBAL_ERRORS_DAYS_FILTER_DECREMENT_CLICKED
+    );
+    setCurrentValue(days);
+  };
+
+  const handleDecrement = () => {
+    if (currentValue === MIN_VALUE) {
+      return;
+    }
+    sendUserActionTrackingEvent(
+      trackingEvents.GLOBAL_ERRORS_DAYS_FILTER_DECREMENT_CLICKED
+    );
+    setCurrentValue(currentValue ? currentValue - 1 : 0);
+  };
+
+  const handleIncrement = () => {
+    if (currentValue === MAX_VALUE) {
+      return;
+    }
+
+    sendUserActionTrackingEvent(
+      trackingEvents.GLOBAL_ERRORS_DAYS_FILTER_INCREMENT_CLICKED
+    );
+    setCurrentValue(currentValue ? currentValue + 1 : 1);
+  };
+
+  const handleApplyClick = () => {
+    sendUserActionTrackingEvent(
+      trackingEvents.GLOBAL_ERRORS_DAYS_FILTER_APPLY_BTN_CLICKED
+    );
+    setSelectedDays(currentValue);
+    setIsDateMenuOpen(false);
+  };
+
+  return (
+    <NewPopover
+      isOpen={isDateMenuOpen}
+      onOpenChange={setIsDateMenuOpen}
+      content={
+        <s.DatePopup>
+          <s.ItemsContainer>
+            <MenuList items={daysFilterMenuItems} highlightSelected={true} />
+            <s.CustomCounterContainer>
+              <s.Counter>
+                <NewIconButton
+                  buttonType="secondaryBorderless"
+                  icon={() => <MinusIcon size={16} />}
+                  onClick={handleDecrement}
+                />
+                <s.CounterInput
```

Comment:
Consider to use input of `type=number` to leverage validation logic on browser:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number